### PR TITLE
Promote etcd frequently asked questions in our bug report template and readme

### PR DESCRIPTION
**Pull request by jmhbnz**
This pull request completes the following subtask of our [refresh for etcd issue triage](https://github.com/etcd-io/etcd/issues/15773):

> Raise the profile of our [frequently asked questions](https://etcd.io/docs/v3.5/faq) page by adding it to key areas like issue templates and readme.

Additionally I have:
- Fixed a number of minor lint errors reported by `markdownlint` in the `README.md`.
- Refactored our bug report template to add an interactive checklist to help guide users to the right channels before opening issues, take a look at a preview below or interactive here: https://github.com/jmhbnz/etcd/issues/new?assignees=&labels=type%2Fbug&projects=&template=bug-report.yml 

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -90,4 +99,4 @@ body:
     attributes:
       label: Relevant log output
       description: Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for backticks.
-      render: shell
+      render: Shell
```
Comment (jmhbnz) [inline .github/ISSUE_TEMPLATE/bug-report.yml:102]: Spec wants this value to be uppercase, refer:
 - https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#attributes-for-textarea
 - https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml#L6383

File: README.md [hunk 2]
```diff
@@ -51,7 +51,6 @@ The easiest way to get etcd is to use one of the pre-built release binaries whic
 For more installation guides, please check out [play.etcd.io](http://play.etcd.io) and [operating etcd](https://etcd.io/docs/latest/op-guide).
 
 [github-release]: https://github.com/etcd-io/etcd/releases
-[branch-management]: https://etcd.io/docs/latest/branch_management
```
Comment (jmhbnz) [inline README.md:54]: This isn't being used in the doc so `markdownlint` wants us to remove it.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+  - type: checkboxes
+    id: confirmations
+    attributes:
+      label: Bug report criteria
+      description: Please confirm this bug report meets the following criteria.
+      options:
+        - label: This bug report is not security related, security issues must be disclosed privately via security@etcd.io.
+        - label: This is not a support request, support requests must be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
+        - label: You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
+        - label: Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.
```
Comment (ahrtr) [inline .github/ISSUE_TEMPLATE/bug-report.yml:16]:
```suggestion
        - label: This bug report is not security related, which should be disclosed privately via security@etcd.io.
        - label: This is not a support request, which should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
        - label: You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
        - label: Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.
```

**Review by serathius — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 3 above)
Comment (serathius) [reply .github/ISSUE_TEMPLATE/bug-report.yml:16]: I think duplication here makes sense as it makes the working stronger and explicit.

Comment (ahrtr) [reply .github/ISSUE_TEMPLATE/bug-report.yml:16]: Please feel free to ignore the duplication. 

I feel like `must` is a little harsh :) , leave to your guys to decide.

Comment (jmhbnz) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: Kept the duplication but switched to `should`.  Middle ground.  We can always tweak it later if we find there are still too many issues being created incorrectly 🙂

**Pull request merged**